zk_toolbox/crates/zk_inception/src/commands/ecosystem/build.rs

```diff
@@ -0,0 +1,215 @@
+    ecosystem_config: &EcosystemConfig,
+    initial_deployment_config: &InitialDeploymentConfig,
+    sender: String,
+) -> anyhow::Result<ContractsConfig> {
+    let spinner = Spinner::new(MSG_INTALLING_DEPS_SPINNER);
+    install_yarn_dependencies(shell, &ecosystem_config.link_to_code)?;
+    build_system_contracts(shell, &ecosystem_config.link_to_code)?;
+    spinner.finish();
+
+    let contracts = build_ecosystem(
+        shell,
+        &mut init_args.ecosystem,
+        init_args.forge_args.clone(),
+        ecosystem_config,
+        initial_deployment_config,
+        sender,
+    )
+    .await?;
+    contracts.save_with_base_path(shell, &ecosystem_config.config)?;
+    Ok(contracts)
+}
+
+async fn build_ecosystem(
+    shell: &Shell,
+    ecosystem: &mut EcosystemArgsFinal,
+    forge_args: ForgeScriptArgs,
+    ecosystem_config: &EcosystemConfig,
+    initial_deployment_config: &InitialDeploymentConfig,
+    sender: String,
+) -> anyhow::Result<ContractsConfig> {
+    if ecosystem.build_ecosystem {
+        return build_ecosystem_inner(
+            shell,
+            forge_args,
+            ecosystem_config,
+            initial_deployment_config,
+            sender,
+        )
+        .await;
+    }
+
+    let ecosystem_preexisting_configs_path =
+        ecosystem_config
+            .get_preexisting_configs_path()
+            .join(format!(
+                "{}.yaml",
+                ecosystem_config.l1_network.to_string().to_lowercase()
+            ));
+
+    // currently there are not some preexisting ecosystem contracts in
+    // chains, so we need check if this file exists.
+    if ecosystem.ecosystem_contracts_path.is_none() && !ecosystem_preexisting_configs_path.exists()
+    {
+        anyhow::bail!(msg_ecosystem_no_found_preexisting_contract(
+            &ecosystem_config.l1_network.to_string()
+        ))
+    }
+
+    let ecosystem_contracts_path =
+        ecosystem
+            .ecosystem_contracts_path
+            .clone()
+            .unwrap_or_else(|| match ecosystem_config.l1_network {
+                L1Network::Localhost => {
+                    ContractsConfig::get_path_with_base_path(&ecosystem_config.config)
+                }
+                L1Network::Sepolia | L1Network::Holesky | L1Network::Mainnet => {
+                    ecosystem_preexisting_configs_path
+                }
+            });
+
+    ContractsConfig::read(shell, ecosystem_contracts_path)
+}
+
+async fn build_ecosystem_inner(
+    shell: &Shell,
+    forge_args: ForgeScriptArgs,
+    config: &EcosystemConfig,
+    initial_deployment_config: &InitialDeploymentConfig,
+    sender: String,
+) -> anyhow::Result<ContractsConfig> {
+    let deploy_config_path = DEPLOY_ECOSYSTEM_SCRIPT_PARAMS.input(&config.link_to_code);
+
+    let default_genesis_config =
+        GenesisConfig::read_with_base_path(shell, config.get_default_configs_path())
+            .context("Context")?;
+
+    let wallets_config = config.get_wallets()?;
```

Comment:
Do you think the wallet config will be correct? 
Probably for build it could be slightly different at least we don't have private keys there